lib/ansible/module_utils/aws/elbv2.py

```diff
@@ -0,0 +1,737 @@
+#!/usr/bin/env python
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Ansible imports
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, get_ec2_security_group_ids_from_names, \
+    ansible_dict_to_boto3_tag_list, boto3_tag_list_to_ansible_dict
+from ansible.module_utils.aws.elb_utils import get_elb, get_elb_listener, convert_tg_name_to_arn
+
+# Non-ansible imports
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass
+import traceback
+import time
+from copy import deepcopy
+
+
+class ElasticLoadBalancerV2(object):
+
+    def __init__(self, connection, module):
+
+        self.connection = connection
+        self.module = module
+        self.changed = False
+        self.new_load_balancer = False
+        self.scheme = module.params.get("scheme")
+        self.name = module.params.get("name")
+        self.subnet_mappings = module.params.get("subnet_mappings")
+        self.subnets = module.params.get("subnets")
+        self.deletion_protection = module.params.get("deletion_protection")
+        self.wait = module.params.get("wait")
+
+        if module.params.get("tags"):
+            self.tags = ansible_dict_to_boto3_tag_list(module.params.get("tags"))
+        else:
+            self.tags = None
+        self.purge_tags = module.params.get("purge_tags")
+
+        self.elb = get_elb(connection, module, self.name)
+        if self.elb is not None:
+            self.elb_attributes = self.get_elb_attributes()
+            self.elb['tags'] = self.get_elb_tags()
+        else:
+            self.elb_attributes = None
+
+    def wait_for_status(self, elb_name, status):
+        polling_increment_secs = 15
+        max_retries = self.module.params.get('wait_timeout') // polling_increment_secs
+        status_achieved = False
+
+        for x in range(0, max_retries):
+            try:
+                response = get_elb(self.connection, self.module, elb_name)
+                if response['State']['Code'] == status:
+                    status_achieved = True
+                    break
+                else:
+                    time.sleep(polling_increment_secs)
+            except (BotoCoreError, ClientError) as e:
+                self.module.fail_json_aws(e, msg=str(e))
```

Comment:
I've noticed that in a few modules already.  I'm not a big fan of the custom message because ClientError could indicate so many things.  Sure, most of the time the error thrown will match up with the custom message but i'm sure there could be cases where the custom message does not really relate to the Exception unless you're very careful.  I can see it causing much confusion.

Anyway, point noted about not requiring if i have no custom message so i will remove